Defra.Cdp.Backend.Api/Endpoints/VanityUrlsEndpoint.cs

```diff
@@ -8,17 +8,37 @@ public static class VanityUrlsEndpoint
 {
     public static void MapVanityUrlsEndpoint(this IEndpointRouteBuilder app)
     {
-        app.MapGet("/vanity-urls/{service}/{environment}", ServiceVanityUrl);
+        app.MapGet("/vanity-urls/{service}/{environment}", ServiceVanityUrls);
+        app.MapGet("/vanity-urls/{service}", AllServiceVanityUrls);
     }
-    
-    private static async Task<IResult> ServiceVanityUrl(
+
+    private static async Task<IResult> ServiceVanityUrls(
         IVanityUrlsService vanityUrlsService,
         string service,
         string environment,
         CancellationToken cancellationToken)
     {
         var result = await vanityUrlsService.FindVanityUrls(service, environment, cancellationToken);
-        return result == null ? Results.NotFound(new ApiError("Not found")) : Results.Ok(new VanityUrlsResponse(result));
+        return result == null
+            ? Results.NotFound(new ApiError("Not found"))
+            : Results.Ok(new VanityUrlsResponse(result));
+    }
+
+    private static async Task<IResult> AllServiceVanityUrls(
+        IVanityUrlsService vanityUrlsService,
+        string service,
+        CancellationToken cancellationToken)
+    {
+        var result = await vanityUrlsService.FindAllVanityUrls(service, cancellationToken);
+
+        if (result.Count == 0)
+        {
+            return Results.NotFound(new ApiError("Not found"));
```

Comment:
assuming its a valid environment, are we not better to just return an empty set?

Reply:
For a service, we're only returning the environments that have a vanity url. We could return all environments and default them to an empty list of vanity urls but this mirrors that actually no vanity urls were found. 

We return a 404 for secrets so thought it better to keep it consistent but think both are equally as valid.